crates/uv/src/commands/pip_list.rs

```diff
@@ -70,6 +79,205 @@ pub(crate) fn pip_list(
         return Ok(ExitStatus::Success);
     }
 
+    if outdated {
+        let constraints: &[RequirementsSource] = Default::default();
+        let overrides: &[RequirementsSource] = Default::default();
+        let extras: ExtrasSpecification<'_> = ExtrasSpecification::default();
+        let connectivity: Connectivity = Connectivity::Online;
+
+        let resolution_mode: ResolutionMode = ResolutionMode::Highest;
+        let prerelease_mode: PreReleaseMode = PreReleaseMode::default();
+        let dependency_mode: DependencyMode = DependencyMode::default();
+        let upgrade: Upgrade = Upgrade::All;
+        let index_locations: IndexLocations = IndexLocations::default();
+        let keyring_provider: KeyringProvider = KeyringProvider::default();
+        let reinstall: Reinstall = Reinstall::All;
+        let setup_py: SetupPyStrategy = SetupPyStrategy::default();
+        let config_settings: ConfigSettings = ConfigSettings::default();
+        let no_build: NoBuild = NoBuild::None;
+        let no_binary: NoBinary = NoBinary::None;
+        let exclude_newer: Option<DateTime<Utc>> = None;
+        let break_system_packages: bool = false;
+        let native_tls: bool = false;
+
+        let _lock = venv.lock()?;
+
+        // Determine the set of installed packages.
+        let site_packages = SitePackages::from_executable(&venv)?;
+
+        let requirements = &site_packages
+            .by_name
+            .keys()
+            .map(|p| RequirementsSource::Package(p.as_dist_info_name().to_string()))
+            .collect::<Vec<_>>();
+
+        // Read all requirements from the provided sources.
+        let spec =
```

Comment:
We crib a large chunk of code from `pip_install` in order to resolve our list of requirements down to a list of packages which are available to install. This is the part where I'm not sure if this is the best approach, or if there's a way to achieve this without so much cribbed/duplicated code.